app/src/main/java/co/kr/soptandroidseminar/SignInActivity.kt

```diff
@@ -0,0 +1,40 @@
+package co.kr.soptandroidseminar
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import co.kr.soptandroidseminar.databinding.ActivitySignInBinding
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignInBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.btnSignin.setOnClickListener {
+            clickLogin()
+        }
+
+        binding.tvSigninSignup.setOnClickListener {
+            clickSignUp()
+        }
+    }
+
+    private fun clickLogin() {
+        if(!binding.etSigninId.text.isNullOrBlank() && !binding.etSigninPw.text.isNullOrBlank()) {
+            Toast.makeText(this, "안녕하세요 ${binding.etSigninId.text}!", Toast.LENGTH_SHORT).show()
+            val intent = Intent(this, HomeActivity::class.java)
+            startActivity(intent)
+        } else {
+            Toast.makeText(this, "ID/PW를 확인해주세요!", Toast.LENGTH_SHORT).show()
+        }
+    }
```

Comment:
선생님 scope 함수 쓰시면 좀 더 깔끔하게 표현 가능할 것 같네요!
```suggestion
    private fun clickLogin() {
    binding.apply {
        if(!etSigninId.text.isNullOrBlank() && !etSigninPw.text.isNullOrBlank()) {
            Toast.makeText(this, "안녕하세요 ${etSigninId.text}!", Toast.LENGTH_SHORT).show()
            val intent = Intent(this, HomeActivity::class.java)
            startActivity(intent)
        } else {
            Toast.makeText(this, "ID/PW를 확인해주세요!", Toast.LENGTH_SHORT).show()
        }
        }
    }
```
대충 이런식으로요!! 스코프 함수는 이외에도 많으니까 한 번 찾아보시길 바랍니다 :)

Reply:
그리고 Toast의 경우 확장함수로 빼서 만들면 더 깔끔할 것 같네요~!

Reply:
`!binding.etSigninId.text.isNullOrBlank() && !binding.etSigninPw.text.isNullOrBlank()` 이 부분도 함수화 시키는 게 가독성 측면에서 좋을 것 같네요~!

Reply:
[https://github.com/29th-WE-SOPT-Android-Part/Android-Songhyeon/pull/1#discussion_r726959991](리뷰)에서 1번 항목 참고 하면 좋을 듯